hibernate-core/src/main/java/org/hibernate/type/TypeFactory.java

```diff
@@ -131,12 +131,13 @@ public Type type(Class<Type> typeClass, Properties parameters) {
 	}
 
 	// todo : can a Properties be wrapped in unmodifiable in any way?
-	private final static Properties EMPTY_PROPERTIES = new Properties();
+	private final static ConcurrentHashMapBackedProperties EMPTY_PROPERTIES = new ConcurrentHashMapBackedProperties();
 
 	public static void injectParameters(Object type, Properties parameters) {
 		if ( ParameterizedType.class.isInstance( type ) ) {
 			if ( parameters == null ) {
-				( (ParameterizedType) type ).setParameterValues( EMPTY_PROPERTIES );
+				ConcurrentHashMapBackedProperties	concurrentParams = new ConcurrentHashMapBackedProperties(parameters);
```

Comment:
Why do you use this new `ConcurrentHashMapBackedProperties` when the `parameters` is null, but not when the `parameters` is not null?